### PR TITLE
add MetaMask ethereum.enable() method

### PR DESCRIPTION
**Pull request by guix77**
Add new MetaMask ethereum.enable() method to approve accounts access if user has turned on privacy option.

Doc :

- https://blog.expo.io/react-native-meets-async-functions-3e6f81111173
- https://github.com/MetaMask/metamask-extension/pull/4703

### CODE REVIEW

**Comment by dangell7** [comment]
Nice Add!!

File: src/web3/web3Saga.js [hunk 1]
```diff
@@ -10,8 +10,25 @@ export function * initializeWeb3 ({ options }) {
   try {
     var web3 = {}
 
-    // Checking if Web3 has been injected by the browser (Mist/MetaMask)
-    if (typeof window.web3 !== 'undefined') {
+    // Checking if Web3 has been injected by Metamask (new version with ethereum.enable() method for privacy).
+    if (window.ethereum) {
+      web3 = new Web3(window.ethereum);
+      // Request account access if needed.
+      yield window.ethereum.enable();
```
Comment (cliffhall) [inline src/web3/web3Saga.js:17]: Is `yield` the right thing here? Everything after this line will not be executed unless the iterator's .next() is called again. In the only place [this generator is referenced](https://github.com/trufflesuite/drizzle/blob/8f30cc9eac0f5d604346499e3450d916919c3e5d/src/drizzleStatus/drizzleStatusSaga.js#L17) it is just called once.

Comment (cliffhall) [reply src/web3/web3Saga.js:17]: In the Metamask[ description of this change](https://medium.com/metamask/https-medium-com-metamask-breaking-change-injecting-web3-7722797916a8), the function is `async` and this call is prefaced with `await`.

Comment (guix77) [reply src/web3/web3Saga.js:17]: Hi,

When looking at the compiled code, it seems to me that .next() is called.

capture-2018-10-21 15 35 47

Line 8066: _context.next=6 then ethereum.enable is called and  next iteration is launched when the yield is finished, context = 6 so in the switch the web3 init continues

Plus: it works!

It seems to me that yield acts just as an await in a generator function.

But I acknoledge it's the first time I use yield so I might be doing something wrong...

**Comment by cliffhall** [comment]
Cool. I only asked because I wasn’t certain and it’s not obvious how the saga figures into call.

**Comment by guix77** [comment]
No problem, it's better to check two times than not !!! Anyways  it's now included along with polling changes (https://github.com/trufflesuite/drizzle/pull/131), so closing this PR.

**Comment by guix77** [comment]
Yup, and thanks for the review ! Anyway it seems to be included in https://github.com/trufflesuite/drizzle/pull/131 now !

**Comment by guix77** [comment]
Yup, and thanks for the review ! Anyway it seems to be included in https://github.com/trufflesuite/drizzle/pull/131 now !

**Pull request closed without merging**